### PR TITLE
Updated Model Classes

### PR DESCRIPTION
**Pull request by german-e-mas**
Addressed the comments from the previous PR. Changes include:
- Renaming the packages to follow the naming conventions.
- Included comments on how the speed and direction are treated.
- Renamed "pos" to "position" in all cases.
- Removed unused setters and getters. Left only the ones I used.
- Updated Position methods. There is no need to individually change x and y, so they got changed.
- Removed unused overwritten`hashCode` and `equals` methods.
- Added a private static ID field in Aircraft class, in order to keep track of the number of instances created.
- Another private ID field, equal to the static ID at the instantiation of the Aircrafts, acts as a each aircrafts' ID.
- Position values change to double.
- Position has Boundary values and a method to check if a Position is inside them.
- Aircraft can check if they are inside the position's boundaries as well.
- Aircrafts get removed if they land or get off the position boundaries.
- A crash terminates the update task and cleans the lists.

ptal @pablisho

### CODE REVIEW

File: app/src/main/java/mas/german/landingplanes/Game.java [hunk 1]
```diff
@@ -75,13 +74,18 @@ public void run() {
                     for (LandingSite site : mSites) {
                         if (aircraft.land(site)) {
                             mScore++;
-                            // I should remove the aircraft once it lands, but this causes an
-                            // exception, and Android Studio is not showing them.
-                            // mAircrafts.remove(aircraft);
+                            synchronized (mAircrafts) {
+                                mAircrafts.remove(aircraft);
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Game.java]: Did you try this? I think you can't remove an item from a list you are iterating using a for loop. You should search for this problem and take a look at Iterator.

File: app/src/main/java/mas/german/landingplanes/Game.java [hunk 2]
```diff
@@ -75,13 +74,18 @@ public void run() {
...
+                            }
                         }
                     }
 
-                    // Todo: Delete aircrafts that are out of bounds.
+                    // Delete aircrafts that are outside the map.
+                    if (aircraft.isOutOfBounds()) {
+                        synchronized (mAircrafts) {
+                            mAircrafts.remove(aircraft);
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Game.java]: same

File: app/src/main/java/mas/german/landingplanes/Position.java [hunk 3]
```diff
@@ -6,6 +6,11 @@
  */
 public class Position {
     private static final String TAG = Position.class.getSimpleName();
+    // Boundaries of the map.
+    private static final double BOUNDARY_LEFT = 0;
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Position.java]: This should be part of the Map class, not position. The position itself doesn't care if it's inside the map or not.

File: app/src/main/java/mas/german/landingplanes/Position.java [hunk 4]
```diff
@@ -26,6 +31,15 @@ public void add(Position position) {
         mY += position.mY;
     }
 
+    public boolean isOutOfBounds() {
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Position.java]: this method should belong to Map class

File: app/src/main/java/mas/german/landingplanes/aircrafts/Aircraft.java [hunk 5]
```diff
@@ -35,6 +35,10 @@ public int getRadius() {
         return mRadius;
     }
 
+    public void setRadius(int radius) {
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/aircrafts/Aircraft.java]: why are you changing the radius after it was created? I think this field should never change.

File: app/src/main/java/mas/german/landingplanes/aircrafts/Aircraft.java [hunk 6]
```diff
@@ -60,6 +64,13 @@ public boolean crashesWith(Aircraft otherAircraft) {
         }
     }
 
+    /**
+     * Check if the Aircraft is farther than the position bounds.
+     */
+    public boolean isOutOfBounds() {
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/aircrafts/Aircraft.java]: Same here, I don;t think this belongs here.

**Comment by german-e-mas** [comment]
I addressed the comments.

ptal again @pablisho

File: app/src/main/java/mas/german/landingplanes/Game.java [hunk 7]
```diff
@@ -28,6 +28,7 @@
 
     private ArrayList<Aircraft> mAircrafts;
     private ArrayList<LandingSite> mSites;
+    private Iterator<Aircraft> mIterator;
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Game.java]: This shouldn't be a field of this class. You should create an iterator each time.

File: app/src/main/java/mas/german/landingplanes/Map.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
+package mas.german.landingplanes;
+
+import mas.german.landingplanes.aircrafts.Aircraft;
+
+public class Map {
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Map.java]: Conceptually I think this class is wrong to have only static methods. Consider doing it a singleton. Also the boundaries shouldn't be static.

**Pull request merged**